Modules/@babylonjs/react-native/shared/BabylonNative.cpp

```diff
@@ -0,0 +1,230 @@
+            m_jsCallInvoker->invokeAsync([this]()
+            {
+                if (m_graphics)
+                {
+                    m_graphics->DisableRendering();
+                }
+            });
+        }
+
+        void SetPointerButtonState(uint32_t pointerId, uint32_t buttonId, bool isDown, uint32_t x, uint32_t y)
+        {
+            if (isDown)
+            {
+                m_nativeInput->PointerDown(pointerId, buttonId, x, y);
+            }
+            else
+            {
+                m_nativeInput->PointerUp(pointerId, buttonId, x, y);
+            }
+        }
+
+        void SetPointerPosition(uint32_t pointerId, uint32_t x, uint32_t y)
+        {
+            m_nativeInput->PointerMove(pointerId, x, y);
+        }
+
+        jsi::Value get(jsi::Runtime& runtime, const jsi::PropNameID& prop) override
+        {
+            if (prop.utf8(runtime) == "initializationPromise")
+            {
+                return { runtime, m_initPromise };
+            }
+            else if (prop.utf8(runtime) == "reset")
+            {
+                return jsi::Function::createFromHostFunction(runtime, prop, 0, [this](jsi::Runtime& rt, const jsi::Value&, const jsi::Value*, size_t) -> jsi::Value
+                {
+                    this->ResetView();
+                    return {};
+                });
+            }
+            else if (prop.utf8(runtime) == "setEngineInstance")
+            {
+                return jsi::Function::createFromHostFunction(runtime, prop, 0, [this](jsi::Runtime& rt, const jsi::Value&, const jsi::Value* args, size_t count) -> jsi::Value
+                {
+                    if (count == 0 || !args[0].isObject())
+                    {
+                        m_disposeEngine = {};
+                    }
+                    else
+                    {
+                        m_disposeEngine = [&rt, engineInstanceValue{ std::make_shared<jsi::Value>(rt, args[0]) }]()
+                        {
+                            auto engineInstance = engineInstanceValue->getObject(rt);
+                            engineInstance.getPropertyAsFunction(rt, "dispose").callWithThis(rt, engineInstance);
+                        };
+                    }
+                    return {};
+                });
+            }
+            
+            return jsi::Value::undefined();
+        }
+        
+    private:
+        jsi::Value m_initPromise{};
+        std::function<void()> m_resolveInitPromise{};
+
+        Napi::Env m_env;
+        std::shared_ptr<facebook::react::CallInvoker> m_jsCallInvoker{};
+
+        std::shared_ptr<bool> m_isRunning{};
+        std::unique_ptr<Graphics> m_graphics{};
+        Plugins::NativeInput* m_nativeInput{};
+
+        std::function<void()> m_disposeEngine{};
+    };
+
+    namespace
+    {
+        constexpr auto JS_INSTANCE_NAME{ "BabylonNative" };
+        std::weak_ptr<ReactNativeModule> g_nativeModule{};
```

Comment:
>g_nativeModule [](start = 41, length = 14)

should this be a map with jsiruntimes as keys? I don't know that we'll ever live in a world where we see multiple jsiruntimes in a single application life cycle, but it seems like a jsiruntime to native module mapping may make more sense than a global/static variable

Reply:
I did that in the previous implementation on Android, and ultimately I thought it made it a lot more complex to cover a scenario that I don't think we'll ever have (multiple JS engine instances existing at the same time). It's not just the map that makes it more complex, it also means the `jsi::Runtime` must be passed to every one of the C APIs that I expose. Again, to me this just seems like added complexity for yagni. If you disagree let me know and I'm happy to explore it further.

Reply:
I don't have a strong preference. I think on the UWP side it was straight forward to initialize per jsiruntime/react context. I have not actually seen any scenarios where multiple jsiruntimes/react contexts are created/used.